From 053e63dc8ab2081bffc68f9f1e534d01c9f5882e Mon Sep 17 00:00:00 2001
From: Ivan Evlogiev <quic_ievlogie@quicinc.com>
Date: Mon, 16 Jan 2023 12:40:03 +0200
Subject: [PATCH] meta-qti-ml: Add Tensorflow Lite Prebuilt recipe

- This recipe can be used to link with the prebuilt tensorflow library.
- It requires to have the tflite-dev.tar.gz prebuilt archive placed at
downloads directory.

Change-Id: I4e14ff63300ab467f00640a4772eebeabc8c2c68
---
 .../gstreamer1.0-plugins-qti-oss-all.bb       |  2 ++
 .../gstreamer1.0-plugins-qti-oss-mltflite.bb  |  2 +-
 .../tensorflow/tensorflow-lite-prebuilt.bb    | 24 +++++++++++++++++++
 3 files changed, 27 insertions(+), 1 deletion(-)
 create mode 100644 recipes/tensorflow/tensorflow-lite-prebuilt.bb

diff --git a/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-all.bb b/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-all.bb
index 8618074..9684ffa 100644
--- a/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-all.bb
+++ b/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-all.bb
@@ -31,6 +31,7 @@ do_compile[depends] = " \
       gstreamer1.0-plugins-qti-oss-mldemux:do_compile \
       ${@bb.utils.contains("MACHINE_FEATURES", "qti-aic", "gstreamer1.0-plugins-qti-oss-mlaic:do_compile", "", d)} \
       ${@bb.utils.contains("DISTRO_FEATURES", "tensorflow-lite", "gstreamer1.0-plugins-qti-oss-mltflite:do_compile", "", d)} \
+      ${@bb.utils.contains("DISTRO_FEATURES", "tensorflow-lite", "tensorflow-lite-prebuilt:do_compile", "", d)} \
       ${@bb.utils.contains("DISTRO_FEATURES", "qti-snpe", "gstreamer1.0-plugins-qti-oss-mlsnpe:do_compile", "", d)} \
       ${@bb.utils.contains("COMBINED_FEATURES", "qti-uvc", "gstreamer1.0-plugins-qti-oss-umd-daemon:do_compile", "", d)} \
       ${@bb.utils.contains_any("COMBINED_FEATURES", "qti-video qti-audio", "gstreamer1.0-omx:do_compile", "", d)} \
@@ -70,6 +71,7 @@ do_package[depends] = " \
       gstreamer1.0-plugins-qti-oss-mldemux:do_package_write_ipk \
       ${@bb.utils.contains("MACHINE_FEATURES", "qti-aic", "gstreamer1.0-plugins-qti-oss-mlaic:do_package_write_ipk", "", d)} \
       ${@bb.utils.contains("DISTRO_FEATURES", "tensorflow-lite", "gstreamer1.0-plugins-qti-oss-mltflite:do_package_write_ipk", "", d)} \
+      ${@bb.utils.contains("DISTRO_FEATURES", "tensorflow-lite", "tensorflow-lite-prebuilt:do_package_write_ipk", "", d)} \
       ${@bb.utils.contains("DISTRO_FEATURES", "qti-snpe", "gstreamer1.0-plugins-qti-oss-mlsnpe:do_package_write_ipk", "", d)} \
       ${@bb.utils.contains("COMBINED_FEATURES", "qti-uvc", "gstreamer1.0-plugins-qti-oss-umd-daemon:do_package_write_ipk", "", d)} \
       ${@bb.utils.contains_any("COMBINED_FEATURES", "qti-video qti-audio", "gstreamer1.0-omx:do_package_write_ipk", "", d)} \
diff --git a/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-mltflite.bb b/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-mltflite.bb
index 7b7ca4f..197d37a 100644
--- a/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-mltflite.bb
+++ b/recipes/gstreamer/gstreamer1.0-plugins-qti-oss-mltflite.bb
@@ -10,7 +10,7 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/${LICENSE};md5=3775480a712fc46a
 DEPENDS := "gstreamer1.0"
 DEPENDS += "gstreamer1.0-plugins-base"
 DEPENDS += "gstreamer1.0-plugins-qti-oss-base"
-DEPENDS += "tensorflow-lite"
+DEPENDS += "tensorflow-lite-prebuilt"
 
 FILESPATH =+ "${WORKSPACE}/vendor/qcom/opensource/gst-plugins-qti-oss/:"
 
diff --git a/recipes/tensorflow/tensorflow-lite-prebuilt.bb b/recipes/tensorflow/tensorflow-lite-prebuilt.bb
new file mode 100644
index 0000000..24c15c4
--- /dev/null
+++ b/recipes/tensorflow/tensorflow-lite-prebuilt.bb
@@ -0,0 +1,24 @@
+SUMMARY = "Tensorflow Lite Prebuilts"
+DESCRIPTION = "TensorFlow Lite C++ Prebuilt Library"
+
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/\
+${LICENSE};md5=89aea4e17d99a7cacdbeed46a0096b10"
+
+SRC_URI = "\
+	file://tflite-dev.tar.gz \
+	"
+
+S = "${WORKDIR}"
+
+do_install_append() {
+
+	cp -r ${S}/usr ${D}/
+
+}
+
+FILES_${PN} = "${libdir}/lib*.so ${bindir}/*"
+FILES_${PN}-dev += "${includedir}"
+
+SOLIBS = ".so*"
+FILES_SOLIBSDEV = ""
-- 
2.17.1

